src/infra/azure/functions/validate-document-upload.ts

```diff
@@ -52,6 +55,11 @@ const validateDocument = makeValidateDocument(
   isDocumentUploadedToBlobStorage
 );
 
+const updateDocumentStatus = makeChangeDocumentStatus(
+  getSignatureRequest,
```

Comment:
It's not clear to me what this method and this one:

```
const validateDocument = makeValidateDocument(
  getSignatureRequest,
  upsertSignatureRequest,
  isDocumentUploadedToBlobStorage
);
```

are supposed to do. It looks like they just get the signature request and save the same content into the db.

Reply:
`makeChangeDocumentStatus` only changes the state of the document on the DB based on the current state, while `makeValidateDocument` takes care of validating the document and then adding the url to the DB. These functions have been separated because before starting the validation with `makeValidateDocument` it is necessary to change the status of the document to `WAIT_FOR_VALIDATION`. Having everything in a single function, the status of the document would change only at the end of the whole validation process.

Reply:
I'm wondering how since by the code it looks they do the upsert just after getting the object from the database (isn't it te same object that gets written?)